### PR TITLE
doc(manual): Update standard library and testing manual pages

### PR DESCRIPTION
**Pull request by cknight**
Updates to the Standard Library and Testing sections of the manual.

EDIT: look like two of us updating the testing docs at the same time. I've replaced my filter flag usage with the update from #6313.

Further edit:  as has been pointed out in the commentary below, the help for the cli flag `--fliter` is not accurate, or at the very least is misleading.  This PR also updates this cli help text.

### CODE REVIEW

File: docs/standard_library.md [hunk 1]
```diff
@@ -55,4 +72,7 @@ To make sure that API producing error is unstable check
 [`lib.deno.unstable.d.ts`](https://github.com/denoland/deno/blob/master/cli/js/lib.deno.unstable.d.ts)
 declaration.
 
-This problem should be fixed in the near future.
+This problem should be fixed in the near future. Please also note that while the
+std library is currently considered unstable, the `--unstable` flag is only
+required when using the std library if the std module itself uses an unstable
+Deno runtime API.
```
Comment (nayeemrmn) [inline docs/standard_library.md:78]: It's best not to bring up these two very different types of "unstable" in the same sentence. There is the first meaning that `std` could have breaking changes whenever, and there is the second meaning that parts of `std` use unstable runtime APIs (which is basically a bug which we'll fix and never allow to happen again).

Also, there's no meaningful segmentation of whether or not an unstable API usage shows up in your dependency graph, it's not something users can easily work around. We should word this more flippantly...
```suggestion
This problem should be fixed in the near future. Feel free to omit the flag if the particular modules you depend on compile successfully without it.
```

**Comment by sebastienfilion** [comment]
Thank you @cknight for this update. I was writing the documentation and then I stumbled on your PR; I would like to request a change: The documentation currently states

> This command will run any test which contains the pattern "hello world" 

I think that the word "pattern" is misleading, it kinda infers that there's a way to do pattern matching of some kind, maybe glob-like or REGEX. Although the filter function accepts, a string or a REGEX, the CLI simply passes along whatever string is parsed from the subcommand's flags.

```ts
// ./cli/js/testing.ts
if (filter) {
      if (filter instanceof RegExp) {
        passes = passes && filter.test(def.name);
      } else {
        passes = passes && def.name.includes(filter);
      }
    }
```

Maybe it would be more accurate to say "which contains the string[...]".
If you agree with this change, maybe we should also change the subcommand's help description for the flag.

```rust
/// ./cli/flags.rs
...
    .arg(
      Arg::with_name("filter")
        .long("filter")
        .takes_value(true)
        .help("A pattern to filter the tests to run by"),
    )
...
```

Otherwise, I think it could be a nice feature to support a REGEX filter from the CLI... Something like:

```shell
deno test --filter "/my-tests\d+/"
```

**Comment by cknight** [comment]
Thanks @sebastienfilion, I agree and have made the change you requested to the manual.  As for the cli flag help, this should ideally be in a separate PR.  Let me know if you want to pick that up, otherwise, I'm happy to do so.

**Comment by cknight** [comment]
@sebastienfilion as a further thought, the code in `cli/js/testing.ts` could check the filter string and if it starts and ends with a `/`, then convert it to RegExp?  Could be a quick easy win, especially if `testing.js` already supports RegExp.

**Comment by sebastienfilion** [comment]
@cknight Yes I agree, it would an easy and isolated fix. 
I created a PR for the flag's description -- if I can add this feature, I will cancel my PR and create a new one.

**Comment by nayeemrmn** [comment]
> As for the cli flag help, this should ideally be in a separate PR.

The manual and cli should be updated together where consistency is wanted. That's why the manual is in the same repo...

**Comment by sebastienfilion** [comment]
> The manual and cli should be updated together where consistency is wanted. That's why the manual is in the same repo...

Thank you @nayeemrmn, I will close my PR.

@cknight Here's a patch if you need it.
```
Index: cli/flags.rs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- cli/flags.rs	(revision 75bb9dbdfc7f8b4e8d17978808ae575e61843aef)
+++ cli/flags.rs	(revision 46f441830875369d3e508ebebe58ac2e924c0a82)
@@ -1053,7 +1053,7 @@
 
 Grant permission to read allow-listed files from disk:
   deno run --allow-read=/etc https://deno.land/std/http/file_server.ts
-  
+
 Deno allows specifying the filename '-' to read the file from stdin.
   curl https://deno.land/std/examples/welcome.ts | target/debug/deno run -",
     )
@@ -1077,7 +1077,7 @@
       Arg::with_name("filter")
         .long("filter")
         .takes_value(true)
-        .help("A pattern to filter the tests to run by"),
+        .help("A string to filter the tests to run by name"),
     )
     .arg(
       Arg::with_name("files")
```

**Comment by cknight** [comment]
> > As for the cli flag help, this should ideally be in a separate PR.
> 
> The manual and cli should be updated together where consistency is wanted. That's why the manual is in the same repo...

Fair enough.  I'll update my PR.

**Pull request merged**